File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/Details.java
package org.firstinspires.ftc.teamcode.util;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.geometry.Pose2d;

import org.firstinspires.ftc.robotcore.external.Telemetry;

public class Details {
    public static Pose2d robotPose = new Pose2d();
    public static Pose2d poseVelocity = new Pose2d();
    public static Alliance alliance = Alliance.NONE;
    public static Side side = Side.NONE;
    public static OpModeType opModeType = OpModeType.NONE;
    public static TelemetryPacket packet = new TelemetryPacket();
    public static Telemetry telemetry;
    public static Pose2d getFuturePose() {
        return robotPose.plus(poseVelocity);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/field/Details.kt
package org.firstinspires.ftc.teamcode.util.field

import org.firstinspires.ftc.teamcode.util.field.OpModeType
import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.geometry.Pose2d
import org.firstinspires.ftc.robotcore.external.Telemetry

object Details {
    @JvmField
    var robotPose = Pose2d()
    @JvmField
    var poseVelocity = Pose2d()
    @JvmField
    var alliance = Alliance.NONE
    @JvmField
    var side = Side.NONE
    @JvmField
    var opModeType = OpModeType.NONE
    @JvmField
    var packet = TelemetryPacket()
    @JvmField
    var telemetry: Telemetry? = null
    val futurePose: Pose2d
        get() = robotPose.plus(poseVelocity)
}